module/programs/helper.js

```diff
@@ -494,21 +502,72 @@ module.exports = class ProgramsHelper {
         }
         queryData.data.startDate ={"$lte": new Date()}
         queryData.data.endDate ={"$gte": new Date()}
+        
         let targetedPrograms = await this.list(
           pageNo,
           pageSize,
           searchText,
           queryData.data,
-          ["name", "externalId","components"]
+          projection
         );
-             
-        if ( targetedPrograms.success && targetedPrograms.data && targetedPrograms.data.data.length > 0) {
+        
+        return resolve({
+          success: true,
+          message: constants.apiResponses.TARGETED_PROGRAMS_FETCHED,
+          data: targetedPrograms.data
+        });
+
+      } catch (error) {
+
+        return resolve({
+          success : false,
+          message : error.message,
+          data : {}
+        });
+
+      }
+
+    })
+  }
+
+  /**
+   * List of programs related to a user.
+   * @method
+   * @name userRelatedProgramsDetails
+   * @param {Array} programIds - program Ids.
+   * @param {String} pageSize - Page size.
+   * @param {String} pageNo - Page no.
+   * @param {String} searchText - search text.
+   * @param {Array} projection - projection.
+   *  @param {String} userId - userId.
+   * @returns {JSON} - List of programs related to the user.
+   */
+
+  static userRelatedProgramsDetails( programIds, pageNo, pageSize, searchText = "", projection, userId ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let findQuery = {
+          "_id": { "$in" : programIds },
```

Comment:
@aks30 named the function userRelatedProgramsDetails because it returns all program details after adding solutions and resourcesStarted keys to response. if you are suggesting a new function name we can change accordingly